pkg/controller/node/nodecontroller_test.go

```diff
@@ -451,6 +455,67 @@ func TestMonitorNodeStatusEvictPods(t *testing.T) {
 	}
 }
 
+// TestCloudProviderNoRateLimit tests that monitorNodes() immediately deletes
+// pods and the node when kubelet has not reported, and the cloudprovider says
+// the node is gone.
+func TestCloudProviderNoRateLimit(t *testing.T) {
+	tmp := nodeExistsInCloudProvider
+	nodeExistsInCloudProvider = func(nc *NodeController, nodeName string) (bool, error) {
```

Comment:
But in this case, there's no good reason for this to not be a member of node controller.